test/package/test_dependency_api.py

```diff
@@ -265,6 +265,14 @@ def import_module(self, module_name):
                 exporter.intern(["foo", "bar"])
                 exporter.save_source_string("my_module", "import foo; import bar")
 
+    def test_invalid_import(self):
+        """An incorrectly-formed import should raise a PackagingError."""
+        buffer = BytesIO()
+        with self.assertRaisesRegex(PackagingError, "attempted relative import"):
```

Comment:
Consider checking that the exception message also contains the right string for `DEPENDENCY_RESOLUTION_FAILED` (i.e. `Dependency resolution failed`).